src/core/gui/inputdevices/StylusInputHandler.cpp

```diff
@@ -82,7 +82,14 @@ auto StylusInputHandler::handleImpl(InputEvent const& event) -> bool {
             this->actionMotion(event);
         }
         XournalppCursor* cursor = xournal->view->getCursor();
-        cursor->setInvisible(false);
+        EraserVisibility visibility = inputContext->getSettings()->getEraserVisibility();
+        if (visibility == ERASER_VISIBILITY_NEVER ||
+            (visibility == ERASER_VISIBILITY_HOVER && this->deviceClassPressed) ||
+            (visibility == ERASER_VISIBILITY_TOUCH && !this->deviceClassPressed)) {
+            cursor->setInvisible(true);
+        } else {
+            cursor->setInvisible(false);
+        }
```

Comment:
This has side effects for the text tool (which also uses the `invisible` property: Using e.g. `ERASER_VISIBILITY_NONE` the text cursor while hovering is invisible (which it normally isn't).

Reply:
Also please check carefully if the `this->deviceClassPressed` state is always correctly detecting if the stylus is hovering or touching (also with stylus button assignments).